test/jdk/java/awt/Frame/FrameRepackTest.java

```diff
@@ -0,0 +1,177 @@
+            added below the red panel.
+            The added panel is blue and has yellow horizontal and vertical scrollbars.
+
+            If you select menu item 'Flip->not visible', the second panel
+            is removed and the frame appears as it did originally.
+
+            You can repeatedly add and remove the second panel in this way.
+            After such an addition or removal, the frame's location on the screen
+            should not have changed, while the size changes to accommodate
+            either just the red panel or both the red and the blue panels.
+
+            If you resize the frame larger, the red panel remains at the
+            top of the frame with its height fixed and its width adjusted
+            to the width of the frame.
+
+            Similarly, if it is present, the blue panel and its yellow scroolbars
+            remain at the bottom of the frame with fixed height and width adjusted
+            to the size of the frame.  But selecting 'visible' or 'not visible'
+            repacks the frame, thereby adjusting its size tightly to its panel(s).
+
+            Upon test completion, click Pass or Fail appropriately.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("FrameRepackTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .testTimeOut(5)
+                .rows(30)
+                .columns(45)
+                .build();
+
+        SwingUtilities.invokeAndWait(() -> {
+            FrameRepack frame = new FrameRepack();
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+
+            frame.setVisible(true);
+        });
+
+        passFailJFrame.awaitAndCheck();
+    }
+
+}
+
+class FrameRepack extends Frame implements ActionListener {
+
+    Panel south;
+
+    public FrameRepack() {
+        super("FrameRepack");
+
+        // create the menubar
+        MenuBar menubar = new MenuBar();
+        this.setMenuBar(menubar);
+        // create the options
+        Menu flip = new Menu("Flip");
+        MenuItem mi;
+        flip.addSeparator();
+        mi = new MenuItem("visible");
+        mi.addActionListener(this);
+        flip.add(mi);
+        mi = new MenuItem("not visible");
+        mi.addActionListener(this);
+        flip.add(mi);
+
+        menubar.add(flip);
+
+        setLayout(new BorderLayout(2, 2));
+        setBackground(Color.green);
+
+        // north panel is always displayed
+        Panel north = new Panel();
+        north.setBackground(Color.red);
+        north.setLayout(new BorderLayout(2, 2));
+        north.add("North", new Label(" This panel is always displayed"));
+        north.add("Center", new Label(" it is a test "));
```

Comment:
```suggestion
        north.add("North", new Label("This panel is always displayed"));
        north.add("Center", new Label("it is a test "));
```
Is the space used to offset the text from the edge?

Reply:
Yeah, it was it. Now the text is flushed to the left. Fine either way.